ukat/mapping/tests/test_diffusion.py

```diff
@@ -88,17 +88,35 @@ def test_fail_to_fit(self):
                      self.mask)
         assert np.abs(mapper.adc.mean()) < 1e-6
 
+    def test_negative_signal(self):
+        gold_standard_adc = [0.000833, 0.000998, 0.0, 0.004852]
+        gold_standard_adc_err = [7.819414e-05, 1.222237e-04, 0.0, 9.935775e-04]
+        negateive_pixel_array = self.pixel_array.copy()
+        negateive_pixel_array[:30, :, :, :] -= 40000
+        mapper = ADC(negateive_pixel_array, self.affine, self.bvals)
+        adc_stats = arraystats.ArrayStats(mapper.adc).calculate()
+        adc_err_stats = arraystats.ArrayStats(mapper.adc_err).calculate()
+        assert np.sum(mapper.adc[:30, :, :]) == 0
+        npt.assert_allclose([adc_stats['mean']['3D'], adc_stats['std']['3D'],
+                             adc_stats['min']['3D'], adc_stats['max']['3D']],
+                            gold_standard_adc, rtol=5e-4, atol=5e-7)
+        npt.assert_allclose([adc_err_stats['mean']['3D'],
+                             adc_err_stats['std']['3D'],
+                             adc_err_stats['min']['3D'],
+                             adc_err_stats['max']['3D']],
+                            gold_standard_adc_err, rtol=5e-3, atol=1e-7)
+
     def test_real_data(self):
         # Gold standard statistics
-        gold_standard_adc = [0.00146, 0.001057, 0.0, 0.005391]
-        gold_standard_adc_err = [0.000128, 0.000143, 0.0, 0.001044]
+        gold_standard_adc = [0.001455, 0.001058, 0.0, 0.005391]
```

Comment:
I guess this is proof that your change works. Negative values are not being used for the fit anymore and that's why these gold_standard measures are now different.

Reply:
Yeah, its pretty much what I expected, some very small changes to the stats because a small number of voxels have been excluded form the fit now.